app/ide-desktop/lib/dashboard/src/components/Loader.tsx

```diff
@@ -2,28 +2,76 @@
  * @file
  * A full-screen loading spinner.
  */
-import * as tw from 'tailwind-merge'
+import * as twv from 'tailwind-variants'
 
 import Spinner, * as spinnerModule from '#/components/Spinner'
 
 /**
  * Props for a {@link Loader}.
  */
-export interface LoaderProps {
+export interface LoaderProps extends twv.VariantProps<typeof STYLES> {
   readonly className?: string
-  readonly size?: number
+  readonly size?: Sizes | number
+  readonly state?: spinnerModule.SpinnerState
 }
 
-const DEFAULT_SIZE = 32
+/**
+ * The possible sizes for a {@link Loader}.
+ */
+export type Sizes = 'large' | 'medium' | 'small'
+
+const STYLES = twv.tv({
+  base: 'animate-appear-delayed flex h-full w-full items-center justify-center duration-200',
+  variants: {
+    minHeight: {
+      full: 'h-full',
+      h6: 'h-6',
```

Comment:
prefer `6` or `h-6` over `h6`

Reply:
IMO h6 is easier to type and it doesn't violate our rules regarding keys in objects. WDYT?

Reply:
makes sense, but considering it's used as a value in all other code i think it's fine to do an eslint-disable here. but of course this is mostly just personal preference, i just think `h6` looks a bit weird